monkestation/code/modules/surgery/robot_chest_repair.dm

```diff
@@ -92,25 +92,23 @@
 /datum/surgery_step/finalize_positronic_restoration/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
 	if(HAS_TRAIT(target, TRAIT_DEFIB_BLACKLISTED))
 		target.emote("buzz")
-		target.visible_message(span_warning("...[target.p_they()] convulses, then goes offline."))
-		return TRUE
-
-	if (target.stat < DEAD)
+		target.visible_message(span_warning("...[target] does not respond. Their personality matrix appears corrupted."))
+		return FALSE
+	if(target.stat < DEAD)
 		target.visible_message(span_notice("...[target] is completely unaffected! Seems like they're already active!"))
 		return FALSE
-
-	target.cure_husk()
-	target.grab_ghost()
-	target.updatehealth()
-
-	if(target.revive())
+	if(target.stat == DEAD && target.health > SYNTH_BRAIN_WAKE_THRESHOLD) // Only revive them if they are of structural integrity to be revived...
+		target.revive()
+		target.cure_husk()
+		target.grab_ghost()
+		target.updatehealth()
 		target.emote("chime")
 		target.visible_message(span_notice("...[target] reactivates, their chassis coming online!"))
-		return FALSE //This is due to synths having some weirdness with their revive.
+		return TRUE
 	else
 		target.emote("buzz")
-		target.visible_message(span_warning("...[target.p_they()] convulses, then goes offline."))
-		return TRUE
+		target.visible_message(span_warning("...[target] convulses, then goes offline. They appear too damaged to reboot.")) // ...Otherwise, they need more repair.
```

Comment:
```suggestion
		target.visible_message(span_warning("...[target] convulses, then goes offline. [target.p_They()] appear too damaged to reboot.")) // ...Otherwise, they need more repair.
```